Extend/Payment_Main.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend;
+
+use \OxidEsales\Eshop\Core\Registry;
+
+/**
+ * Class Payment_Main
+ *
+ */
+class Payment_Main extends Payment_Main_parent
+{
+
+    /**
+     * Executes parent::render(), creates a currency array
+     *
+     * @return string
+     *
+     */
+    public function render()
+    {
+        $sTemplate = parent::render();
+        $oConfig = Registry::getConfig();
+        if ($this->_aViewData["oxid"] == "wdcreditcard") {
```

Comment:
I suggest getting the payment method name, then having a getter in each payment method where you can get the "additional fields" and in the abstract payment_method you just return an empty array. Then here all you do is merge _aViewData with the getter function

Reply:
This is addressed in Pull Request #32 . Let's prioritise that one and keep this open until #32 is merged.